src/Database/Dialect/SqlserverDialectTrait.php

```diff
@@ -325,6 +325,10 @@ public function rollbackSavePointSQL($name)
      */
     public function newCompiler()
     {
+        if (isset($this->_config['compiler'])) {
+            $class = $this->_config['compiler'] . 'Compiler';
```

Comment:
Maybe throw a detailed exception if the compiler is missing (i.e. missing controllers, etc.)? 

Not sure what others think.